packages/api/src/schema/types/shop.js

```diff
@@ -4,13 +4,24 @@ export default [
       href: String
       title: String
     }
+
+    type CustomEntityInterface {
+      entityName: String!
+      inlineFragment: String!
+    }
+
+    type AdminUiConfig {
+      customProperties: [CustomEntityInterface!]!
+    }
+
     type Shop @cacheControl(maxAge: 180) {
       _id: ID!
       language: Language
       country: Country
       version: String
       userRoles: [String!]!
       externalLinks: [Link]!
+      adminUiConfig: AdminUiConfig
```

Comment:
You're doing extra work to ensure that adminUiConfig is set to an object with an empty array but in GraphQL you define adminUiConfig as optional, i think it should be AdminUiConfig! here